lib/restaurant.rb

```diff
@@ -0,0 +1,35 @@
+class Restaurant
+  def view_menu
+    display.view_menu(menu)
+  end
+
+  def add(meal)
+    choice = meal.downcase.to_sym
+    return display.invalid(meal) unless menu.include?(choice)
+
+    order << menu.slice(choice)
+    display.confirm_added(meal)
+  end
+
+  def view_order
+    display.view_order(order)
+  end
+
+  def place_order
+    puts view_order
+    display.confirm_order
+    order.clear
+    customer_service.confirmation_text
+  end
+
+  private
+
+  attr_reader :menu, :order, :display, :customer_service
+
+  def initialize(args = {})
```

Comment:
What is the reason for having the initialize method be private?

Reply:
it's something that an Exercism mentor told me - that if I have private methods in a class I should put initialize there too, else it's misleading. Is that something you agree with?

Reply:
Interesting! I found [this stackoverflow](https://stackoverflow.com/a/36491066) answer interesting. It looks like you don't need to put the initialize method under the private heading for it to still be private. According to [this style guide](https://github.com/rubocop-hq/ruby-style-guide#classes--modules) the initialize method should go towards the top of the class and not under the private heading

Reply:
Thanks that's good to know! I was previously putting at the top of the class as I knew it was  private by default but was told it's best practice to explicitly show it's private. This makes more sense to me though, having it at the top between class methods and instance methods. Thanks for the tip 🙂